boards/metro_m4/examples/adc.rs

```diff
@@ -27,35 +32,14 @@ fn main() -> ! {
         &mut peripherals.OSCCTRL,
         &mut peripherals.NVMCTRL,
     );
-    let mut pins = hal::Pins::new(peripherals.PORT);
+    let pins = bsp::Pins::new(peripherals.PORT);
     let mut delay = hal::delay::Delay::new(core.SYST, &mut clocks);
     let mut adc0 = Adc::adc0(peripherals.ADC0, &mut peripherals.MCLK, &mut clocks, GCLK11);
-    let mut a0 = pins.a0.into_function_b(&mut pins.port);
-
-    let gclk0 = clocks.gclk0();
-    let rx: Sercom3Pad1<_> = pins
-        .d0
-        .into_pull_down_input(&mut pins.port)
-        .into_pad(&mut pins.port);
-    let tx: Sercom3Pad0<_> = pins
-        .d1
-        .into_pull_down_input(&mut pins.port)
-        .into_pad(&mut pins.port);
-    let uart_clk = clocks
-        .sercom3_core(&gclk0)
-        .expect("Could not configure sercom3 clock");
-
-    let mut uart = UART3::new(
-        &uart_clk,
-        9600.hz(),
-        peripherals.SERCOM3,
-        &mut peripherals.MCLK,
-        (rx, tx),
-    );
+    let mut a0 = pins.a0.into_alternate::<B>();
```

Comment:
Could use the `aliases` feature for this one:
```rust
let mut a0: bsp::Analog0 = pins.a0.into();
```

and `lib.rs` would have something like:
```rust
    PA02 {
         /// Analog pin 0.  Can act as a true analog output
         /// as it has a DAC (which is not currently supported
         /// by this hal) as well as input.
         name: a0,
        aliases: {
            AlternateB: Analog0
     }
```